plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -0,0 +1,82 @@
+import ast
+import re
+from operator import itemgetter
+
+import insightconnect_plugin_runtime
+from typing import Dict, List
+
+from insightconnect_plugin_runtime.exceptions import PluginException
+from insightconnect_plugin_runtime.helper import get_time_hours_ago
+
+from .schema import MonitorSiemLogsInput, MonitorSiemLogsOutput, MonitorSiemLogsState, Component
+from ...util.event import EventLogs
+
+
+class MonitorSiemLogs(insightconnect_plugin_runtime.Task):
+    NEXT_TOKEN = "next_token"  # nosec
+    HEADER_NEXT_TOKEN = "mc-siem-token"  # nosec
+    STATUS_CODE = "status_code"  # nosec
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_siem_logs",
+            description=Component.DESCRIPTION,
+            input=MonitorSiemLogsInput(),
+            output=MonitorSiemLogsOutput(),
+            state=MonitorSiemLogsState(),
+        )
+
+    def run(self, params={}, state={}) -> (List[Dict], Dict):
+        if not state:
+            self.logger.info("First run")
+        else:
+            self.logger.info("Subsequent run")
+            params["token"] = state.get(self.NEXT_TOKEN)
+
+        has_more_pages = False
+
+        while True:
+            try:
+                output, headers, status_code = self.connection.client.get_siem_logs(params)
+            except PluginException as error:
+                state[self.STATUS_CODE] = self._get_status_code(error)
+                return [], state, has_more_pages
+            header_next_token = headers.get(self.HEADER_NEXT_TOKEN)
+            params["token"] = header_next_token
+            if not output:
+                break
+
+            output = self._filter_and_sort_recent_events(output)
+            if len(output) > 0:
+                break
+
+        if header_next_token:
+            state[self.NEXT_TOKEN] = headers.get(self.HEADER_NEXT_TOKEN)
+            state[self.STATUS_CODE] = status_code
+            has_more_pages = True
+
+        return output, state, has_more_pages
+
+    def _filter_and_sort_recent_events(self, output: List[dict]) -> List[dict]:
+        """
+        Filters and sorts a list of events to retrieve only the recent events.
+
+        Args:
+            events (list): A list of event objects to be filtered and sorted.
+
+        Returns:
+            list: The filtered and sorted list of recent events.
+
+        """
+        output = [EventLogs(data=event) for event in output]
+        return sorted(
+            [event.get_dict() for event in output if event.compare_datetime(get_time_hours_ago(hours_ago=24))],
+            key=itemgetter("datetime"),
+        )
+
+    def _get_status_code(self, error: str) -> int:
+        """
+        Extracts the status code from an error message.
+        """
+        pattern = r"'status': \s*(\d+)"
+        return re.search(pattern, error.data).group(1)
```

Comment:
I'm a little weary of using regex to extract status codes. Can you do something similar to what was done in the diff in this task PR (https://github.com/rapid7/insightconnect-plugins/pull/1764/commits/84171e3effdcdc882519d82d68446942559be628)? It's using classes with properties to better structure things for consumption and should make things more reliable/safe.